### PR TITLE
RSSFE-25 fix: fix bugs

### LINKED ISSUE #76: RSSFE-25 fix bugs (currCourse)
Add current course to local storage
case 1: updating cache when it is the first group
case 2: default pagination - page without teams
case 3: event handler on edit button
case 3.1: delete console.log "Info notification"

### PR DESCRIPTION
**Pull request by besovadevka**
Close #76
Added current course to local storage
Fixed pagination, updating cache after mutations

### CODE REVIEW

File: src/components/App/index.tsx [hunk 1]
```diff
@@ -37,8 +37,15 @@ export const App: FC = () => {
     if (!!whoAmI) {
       dispatch({ type: SET_USER_DATA, payload: whoAmI });
     }
-    if (whoAmI?.courses[0])
-      dispatch({ type: SET_CURR_COURSE, payload: whoAmI?.courses[0] });
+    if (whoAmI?.courses[0]) {
+      if (!localStorage.getItem('currCourse')) {
```
Comment (manuminsk) [inline src/components/App/index.tsx:41]: I'd recommend using currentCourse instead of currCourse

File: src/components/App/index.tsx [hunk 2]
```diff
@@ -37,8 +37,15 @@ export const App: FC = () => {
...
+        localStorage.setItem('currCourse', JSON.stringify(whoAmI?.courses[0]));
+      }
+      const currCourse = JSON.parse(
+        localStorage.getItem('currCourse') as string
```
Comment (manuminsk) [inline src/components/App/index.tsx:45]: Could you save 'currCourse' as a const and use it where it's necessary?

File: src/components/Header/components/MenuWrapper/components/CoursesSelect/index.tsx [hunk 3]
```diff
@@ -31,6 +31,7 @@ export const CoursesSelect: FC<CoursesSelectProps> = ({
 
   const onCourseChange = (course: Course) => {
     setDisplayCoursesList(false);
+    localStorage.setItem('currCourse', JSON.stringify(course));
```
Comment (manuminsk) [inline src/components/Header/components/MenuWrapper/components/CoursesSelect/index.tsx:34]: as above => const

File: src/modules/EditProfile/index.tsx [hunk 4]
```diff
@@ -90,6 +90,17 @@ export const EditProfile: FC = () => {
     if (userCourses.length) {
       updateUser().then((data) => {
         dispatch({ type: SET_USER_DATA, payload: data.data.updateUser });
+        if (userData.courses.length !== data.data.updateUser.courses.length) {
+          const newCurrCourse = data.data.updateUser.courses.find(
+            (course: Course) => course.id === data.data.updateUser.courseIds[0]
+          );
+          localStorage.setItem('currCourse', JSON.stringify(newCurrCourse));
```
Comment (manuminsk) [inline src/modules/EditProfile/index.tsx:98]: Const

**Review by manuminsk — APPROVED** [review]

File: src/components/App/index.tsx (hunk 1 above)
Comment (besovadevka) [reply src/components/App/index.tsx]: Done

File: src/components/App/index.tsx (hunk 2 above)
Comment (besovadevka) [reply src/components/App/index.tsx]: Done

File: src/components/Header/components/MenuWrapper/components/CoursesSelect/index.tsx (hunk 3 above)
Comment (besovadevka) [reply src/components/Header/components/MenuWrapper/components/CoursesSelect/index.tsx]: Done

File: src/modules/EditProfile/index.tsx [hunk 5]
```diff
@@ -91,6 +96,17 @@ export const EditProfile: FC = () => {
     if (userCourses.length) {
       updateUser().then((data) => {
```
Comment (MadaShindeInai) [inline src/modules/EditProfile/index.tsx:97]: destructure update user there like. `updateUser().then(({data:{updateUser}) => {`

File: src/modules/StudentsTable/components/Dashboard/components/TableBody/index.tsx [hunk 6]
```diff
@@ -1,29 +1,37 @@
...
       users.map((user: User, index: number) => {
         return [
           `${index + 1 + page * USERS_PER_PAGE}`,
           `${user.firstName} ${user.lastName || null}`,
           user.score,
-          user.teams.length
-            ? (user.teams as Array<Team>)
-                .map((team: Team) => `${team.number}`)
-                .join(', ')
+          (user.teams as Team[]).find(
```
Comment (MadaShindeInai) [inline src/modules/StudentsTable/components/Dashboard/components/TableBody/index.tsx:27]: do we still need this extra description? `(user.teams as Team[])`

Comment (MadaShindeInai) [reply src/modules/StudentsTable/components/Dashboard/components/TableBody/index.tsx:27]: It we have problem - we should may be remove |[] from User type teams field description

**Review by MadaShindeInai — CHANGES_REQUESTED** [review]

File: src/modules/StudentsTable/components/Dashboard/components/TableBody/index.tsx (hunk 6 above)
Comment (besovadevka) [reply src/modules/StudentsTable/components/Dashboard/components/TableBody/index.tsx]: Done

**Review by MadaShindeInai — APPROVED** [review]

**Pull request merged**